Update fit_kwargs_by_estimator example in Task-Oriented-AutoML.md (#561)
microsoft/FLAML: website/docs/Use-Cases/Task-Oriented-AutoML.md

### Extra fit arguments

Extra fit arguments that are needed as one of the argument for model.fit(). The following code shows how you can pass the extra fit argument using the `fit_kwargs_by_estimator` argument in AutoML.fit():

```python
from flaml import AutoML

> from pandas.tests.extension.conftest import as_frame

Reviewer: ```suggestion
```

from sklearn.datasets import load_iris

> X_train, y_train = load_iris(return_X_y=True, as_frame=as_frame)

Reviewer: ```suggestion
X_train, y_train = load_iris(return_X_y=True)
```

automl = AutoML()
automl_settings = {
    "time_budget": 2,
    "task": "classification",
    "log_file_name": "test/iris.log",
    "estimator_list": ["rf"],
    "max_iter": 0,
}

> automl_settings["fit_kwargs_by_estimator"] = {

Reviewer: This example is not good because it has only one estimators and sample_weight is not specific to "rf". The point for this example is that we can pass estimator-specific fit_kwargs.

Reviewer: This comment is not resolved.

    "rf": {

>         "sample_weight":  [1,2,3]

Reviewer: ```suggestion
        "sample_weight": [1,2,3]
```

> Extra fit arguments that are needed by the estimators can be passed to `AutoML.fit()`. For example, if there is a weight associated with each training example, they can be passed via `sample_weight`. For another example, `period` can be passed for time series forecaster. For any extra keywork argument passed to `AutoML.fit()` which has not been explicitly listed in the function signature, it will be passed to the underlying estimators' `fit()` as is.

Reviewer: "gpu_per_trial" should be talked about in this paragraph.

### Extra fit arguments

Extra fit arguments that are needed by the estimators can be passed to `AutoML.fit()`. For example, if there is a weight associated with each training example, they can be passed via `sample_weight`. For another example, `period` can be passed for time series forecaster. For any extra keywork argument passed to `AutoML.fit()` which has not been explicitly listed in the function signature, it will be passed to the underlying estimators' `fit()` as is. For another example, you can set the number of gpus used by each trial with the `gpu_per_trial` argument, which is only used by TransformersEstimator and XGBoostSklearnEstimator. 

In addition, you can specify the different arguments needed by different estimators using the `fit_kwargs_by_estimator` argument. For example, you can set the custom arguments for a Transformers model:

```python
from flaml.data import load_openml_dataset
from flaml import AutoML

X_train, X_test, y_train, y_test = load_openml_dataset(dataset_id=1169, data_dir="./")

automl = AutoML()
automl_settings = {
    "task": "classification",
    "time_budget": 10,

>     "estimator_list": ["catboost"],

Reviewer: Don't include a single estimator.

    "fit_kwargs_by_estimator": {
        "catboost": {

>             "verbose": False,  # setting the verbosity of xgboost to True

Reviewer: ```suggestion
            "verbose": True,  # setting the verbosity of catboost to True
```
